client/components/backup-retention-management/retention-options/retention-options-control.tsx

```diff
@@ -28,10 +30,14 @@ const RetentionOptionsControl: React.FC< RetentionOptionsControlProps > = ( {
 
 	return (
 		<div className="retention-form__options">
-			{ Object.values( retentionOptions ).map( ( option ) => (
+			{ retentionOptions.map( ( option ) => (
 				<RetentionOptionCard
-					{ ...option }
-					key={ option.value }
+					value={ option.id }
+					spaceNeededInBytes={ option.spaceNeededInBytes }
+					upgradeRequired={ option.upgradeRequired }
+					checked={ retentionSelected === option.id }
+					isCurrentPlan={ currentRetentionPlan === option.id }
+					key={ option.id }
```

Comment:
I'd like to point out that using the value as a key might have unexpected consequences and cause unexpected behavior. 

That can mean that if the value changes, the key will also change, and React will be unable to reuse the component, creating a new one. It can also mean that React might unintentionally reuse an old component that had the same value (respectively, key) as the one we're now about to render.

Reply:
Given that for this specific scenario, the `id` is the unique identifier for that option (which only can be 7, 30, 120 or 365 for now), I don't see how this could be an issue. What do you think?

Reply:
I don't see it problematic in this particular scenario, but it is a red flag because someone might borrow this code later and use it with a larger amount of (or infinite) options. It partially defies the idea of how React keys are supposed to serve.

That being said, feel free to add a prefix to the `key` so it's at least different from the value. Adding a comment that this is fine because we're working with a small set of option IDs could also help our future selves.

Reply:
Done!